### ISSUE TITLE
Nav offcanvas set the correct `aria-posinset` for the appender

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47024]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1376732975 by @talldan:

> the appender doesn't have the correct aria-posinset value (it's a minor thing as this isn't announced, but it'd be nice to fix it).
